src/site/_includes/codelab.njk

```diff
@@ -12,11 +12,11 @@ pageScripts:
     {# Use first found path as a default path. #}
     {% set path = paths[postPaths[0]] %}
   {% endif %}
-  <web-codelab class="web-codelab" glitch="{{ glitch }}" {% if glitch_path %}path="{{ glitch_path }}"{% endif %} {% if site.env === 'test' %}testmode{% endif %}>
+  <web-codelab class="web-codelab" glitch="{{ glitch }}" {% if glitch_path %}path="{{ glitch_path }}"{% endif %} {% if site.percy === 'true' %}percymode{% endif %}>
```

Comment:
Instead of adding whole logic of percymode to the component, you can simply not render it here if site.percy === 'true'. It is not the omponent role to understand the test environment limitations.